src/vue/pages/IncomingWithdrawalRequests.vue

```diff
@@ -0,0 +1,368 @@
+  components: {
+    Loader,
+    Drawer,
+    NoDataMessage,
+    CollectionLoader,
+    WithdrawalRequestDetails,
+    EmailGetter,
+  },
+
+  data: _ => ({
+    requestsHistory: [],
+    isHistoryLoaded: false,
+    isLoadingFailed: false,
+    isRequestRejected: false,
+    isDetailsDrawerShown: false,
+    selectedIndex: -1,
+    firstPageLoader: () => {},
+  }),
+  computed: {
+    ...mapGetters({
+      accountId: vuexTypes.accountId,
+    }),
+    selectedRequest () {
+      return this.requestsHistory[this.selectedIndex]
+    },
+  },
+
+  created () {
+    this.initFirstPageLoader()
+  },
+
+  methods: {
+    initFirstPageLoader () {
+      this.isHistoryLoaded = false
+      this.requestsHistory = []
+      this.firstPageLoader = this.getFirstPageLoader(this.accountId)
+    },
+
+    closeDetailsDrawer () {
+      this.isDetailsDrawerShown = false
+    },
+
+    getFirstPageLoader (accountId) {
+      return function () {
+        const endpoint = `/${HORIZON_VERSION_PREFIX}/create_withdraw_requests`
+        return Api.api.getWithSignature(endpoint, {
+          filter: {
+            reviewer: accountId,
+          },
+          include: ['request_details'],
+        })
+      }
+    },
+
+    async setHistory (data) {
+      this.requestsHistory = await Promise.all(
+        data
+          .map(request => new WithdrawalDetailsRequestRecord(request))
+          .map(async request => {
+            request.asset = await this.getAssetFromBalanceId(request)
+            return request
+          })
+      )
+      this.isHistoryLoaded = true
+    },
+
+    async extendHistory (data) {
+      this.requestsHistory = this.requestsHistory.concat(
+        await Promise.all(
+          data
+            .map(request => new WithdrawalDetailsRequestRecord(request))
+            .map(async request => {
+              request.asset = await this.getAssetFromBalanceId(request)
+              return request
+            })
+        )
+      )
+    },
+
+    async getAssetFromBalanceId (request) {
+      const { _rawResponse: token } = await Api.api
+        .get(`balances/${request.balanceId}/asset`)
+      return token.data.code
+    },
+
+    showRequestDetails (index) {
+      this.selectedIndex = index
+      this.isDetailsDrawerShown = true
+    },
+
+    async approveRequest () {
+      this.isRequestRejected = true
+      try {
+        const action = base.xdr.ReviewRequestOpAction.approve().value
+        await this._reviewWithdraw({ action }, this.selectedRequest)
+        await this.rewriteWithdrawalRequest()
+
+        Bus.success('incoming-withdrawal-requests.request-approved-msg')
+      } catch (e) {
+        ErrorHandler.process(e)
+      }
+      this.isRequestRejected = false
+    },
+
+    async rejectRequest (rejectReason) {
+      this.isRequestRejected = true
+      try {
+        const action = base.xdr.ReviewRequestOpAction.permanentReject().value
+        /* eslint-disable-next-line max-len */
```

Comment:
For similar cases use `eslint-disable-next-line` instead. But here you can just make this one:

```js 
await this._reviewWithdraw(
   { action, reason: rejectReason },
   this.selectedRequest
)
```

Reply:
resolved